test/components/views/dialogs/AccessSecretStorageDialog-test.tsx

```diff
@@ -15,113 +15,97 @@ limitations under the License.
 */
 
 import React from "react";
-// eslint-disable-next-line deprecate/import
-import { mount, ReactWrapper } from "enzyme";
-import { act } from "react-dom/test-utils";
 import { IPassphraseInfo } from "matrix-js-sdk/src/crypto/api";
+import { act, fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { MatrixClient } from "matrix-js-sdk/src/matrix";
+import { mocked } from "jest-mock";
 
-import { findByAttr, getMockClientWithEventEmitter, unmockClientPeg } from "../../../test-utils";
-import { findById, flushPromises } from "../../../test-utils";
+import { getMockClientWithEventEmitter, mockPlatformPeg } from "../../../test-utils";
 import AccessSecretStorageDialog from "../../../../src/components/views/dialogs/security/AccessSecretStorageDialog";
 
+const securityKey = "EsTc WKmb ivvk jLS7 Y1NH 5CcQ mP1E JJwj B3Fd pFWm t4Dp dbyu";
+
 describe("AccessSecretStorageDialog", () => {
-    const mockClient = getMockClientWithEventEmitter({
-        keyBackupKeyFromRecoveryKey: jest.fn(),
-        checkSecretStorageKey: jest.fn(),
-        isValidRecoveryKey: jest.fn(),
-    });
+    let mockClient: MatrixClient;
+
     const defaultProps = {
         onFinished: jest.fn(),
         checkPrivateKey: jest.fn(),
         keyInfo: undefined,
     };
-    const getComponent = (props = {}): ReactWrapper =>
-        mount(<AccessSecretStorageDialog {...defaultProps} {...props} />);
 
-    beforeEach(() => {
-        jest.clearAllMocks();
-        mockClient.keyBackupKeyFromRecoveryKey.mockReturnValue("a raw key" as unknown as Uint8Array);
-        mockClient.isValidRecoveryKey.mockReturnValue(false);
+    const renderComponent = (props = {}): void => {
+        render(<AccessSecretStorageDialog {...defaultProps} {...props} />);
+    };
+
+    const enterSecurityKey = async (placeholder = "Security Key"): Promise<void> => {
+        await act(async () => {
```

Comment:
Why is this async?

Reply:
This is a leftover. There was a flush promises inside the function.
→ `async` removed